cpp/src/binaryop/compiled/binary_ops.cuh

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "binary_ops.hpp"
+#include "operation.cuh"
+
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/detail/utilities/integer_utils.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+namespace cudf {
+namespace binops {
+namespace compiled {
+
+template <typename BinaryOperator>
+constexpr bool is_bool_result()
+{
+  return std::is_same_v<BinaryOperator, ops::NullEquals> or
+         std::is_same_v<BinaryOperator, ops::Equal> or
+         std::is_same_v<BinaryOperator, ops::NotEqual> or
+         std::is_same_v<BinaryOperator, ops::Less> or
+         std::is_same_v<BinaryOperator, ops::Greater> or
+         std::is_same_v<BinaryOperator, ops::LessEqual> or
+         std::is_same_v<BinaryOperator, ops::GreaterEqual> or
+         std::is_same_v<BinaryOperator, ops::LogicalOr> or
+         std::is_same_v<BinaryOperator, ops::LogicalAnd>;
```

Comment:
This seems like it would be better implemented using something based off of `invoke_result_t` instead of explicitly listing these operators. Otherwise if a new op is added with a bool output type, you have to remember to go update this list. Ideally there should only be a single source of truth.